src/sentry/utils/event_frames.py

```diff
@@ -63,17 +63,27 @@ def java_frame_munger(frame: EventFrame) -> str | None:
 
 
 def java_new_logic_frame_munger(frame: EventFrame) -> str | None:
+    stacktrace_path = None
     if not frame.module or not frame.abs_path:
         return None
 
-    from sentry.issues.auto_source_code_config.code_mapping import get_path_from_module
+    from sentry.issues.auto_source_code_config.code_mapping import (
+        DoesNotFollowJavaPackageNamingConvention,
+        get_path_from_module,
+    )
 
     try:
         _, stacktrace_path = get_path_from_module(frame.module, frame.abs_path)
+    except DoesNotFollowJavaPackageNamingConvention:
```

Comment:
This handles the exception from [SENTRY-3T8M](https://sentry.sentry.io/issues/6577846344/)

There's a new test case to verify it works as expected.